modules/crawler/src/test/scala/DownloaderTest.scala

```diff
@@ -0,0 +1,27 @@
+package fide
+package crawler
+package test
+
+import cats.effect.IO
+import fide.domain.Federation
+import fide.types.FederationId
+import org.http4s.ember.client.EmberClientBuilder
+import org.typelevel.log4cats.Logger
+import weaver.*
+
+object DownloaderTest extends SimpleIOSuite:
+
+  given Logger[IO] = org.typelevel.log4cats.noop.NoOpLogger[IO]
+
+  test("No new federations after downloads"):
+    EmberClientBuilder
+      .default[IO]
+      .build
+      .use: client =>
+        Downloader(client).fetch
+          .map(x => x._2)
+          .fold[Set[FederationId]](Set.empty)((acc, x) => acc ++ x.map(_.id))
```

Comment:
Is this

~~~scala
.mapFilter(x => x._2.map(_.id))
.to(Set)
~~~

less preferable?

Reply:
This change also eliminates `.compile.last` part, I guess

Reply:
couldn't use `mapFilter`, probably `IO` is not a `FunctorFilter`, so this is the best I came up with: https://github.com/lenguyenthanh/fide/pull/254/commits/4e0665210b63dec313aa322bf614cafb64b12c56. thanks again, really love your reviews!

Reply:
Yep, looks great 👍🏻 (I haven't been sure if `.to(Set)` works). But I suppose `mapFilter` works here, because [there is no bound](https://github.com/typelevel/fs2/blob/8a67a3b4899ee43a9efa642e5f362d2f96be5dbd/core/shared/src/main/scala/fs2/Stream.scala#L5685-L5698) for `F[_]` for `Stream[F[_], A].mapFilter`, [scastie for reference](https://scastie.scala-lang.org/uF2A79MJSwiw6p6yQDuClQ)

Reply:
oh, it was me, I forgot to import `cats.syntax.all.*` 🤦

Reply:
fixed https://github.com/lenguyenthanh/fide/pull/267/files